src/messaging.cc

```diff
@@ -283,10 +283,7 @@ bool MessageEncoder::encode_byte_array(ByteArray* object) {
     too_many_externals_ = true;
     return false;
   }
-  if (encoding_for_size()) {
-    externals_[externals_count_] = object;
-  }
-  externals_count_++;
+  externals_[externals_count_++] = object;
```

Comment:
If I understand correctly, the bug is that the condition is not inverted. correct?
It should have been `if (!encoding_for_size())`